### PR TITLE
[HIP][COMGR] Use COMGR by default for HIP backend.

### PR DESCRIPTION
**Pull request by atamazov**

### CODE REVIEW

**Comment by junliume** [comment]
@atamazov  should we upgrade CI to ROCm 4.5 before or after this PR? The label is urgent. If we are talking about absolute path issue, we are getting it deferred anyway.

**Comment by atamazov** [comment]
As far as I understand, only base installation should be upgraded. No changes in the Dockerfile required.

**Comment by atamazov** [comment]
Blocks 5.1

**Comment by junliume** [comment]
@atamazov is this PR still been blocked by #1257 ?

**Comment by atamazov** [comment]
Yes.

**Comment by atamazov** [comment]
#1257 is not a blocker anymore.

**Comment by atamazov** [comment]
Hmm, it seems like we need both (2) and (3) from https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1257#issuecomment-972820638 to get around the issue.

**Comment by atamazov** [comment]
> #1257 is not a blocker anymore.

Maybe I am wrong. Why: Some CI nodes were upgraded to 5.11.32, and therefore #1257 is hardly reproducible.

I've disabled hpe-rack-15 (with 5.11.32) for a while. Please note that after that we have only one functional node in CI; other MI200 nodes are disabled due to issues.

**Comment by atamazov** [comment]
All the issues resolved, CI tests passed. Please review.

**Comment by atamazov** [comment]
If we do not want this to be included into 5.0, then let's keep this on hold until the release branch is cut.

**Comment by atamazov** [comment]
@junliume The PR includes:

> [COMGR] Fix: Don't remove "-mllvm ..." fetched from hip::device properties. Split before passing to COMGR.

I am going to create a separate PR with this fix in order to include it into 5.0. This is not very important because there is no known issues related to this, but let's better be proactive.

File: CMakeLists.txt
Comment (shurale-nkn) [inline CMakeLists.txt:293]: I would prefer to solve this problem in OpenCL and NOGPU handlers
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/72f06aa8cb53c30ac2f87d34378925ff9401f4ed/CMakeLists.txt#L155-L158
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/72f06aa8cb53c30ac2f87d34378925ff9401f4ed/CMakeLists.txt#L223-L226

**Review by shurale-nkn — APPROVED** [review]
LGTM

File: CMakeLists.txt
Comment (atamazov) [reply CMakeLists.txt:293]: If I do so, then I will need to first set MIOPEN_USE_COMGR_DEFAULT to TRUE and then change it to FALSE in some cases. This makes it harder to track the value of MIOPEN_USE_COMGR_DEFAULT. Do you agree?

Comment (shurale-nkn) [reply CMakeLists.txt:293]: I suppose you can set MIOPEN_USE_COMGR to FALSE in these two cases, instead of changing MIOPEN_USE_COMGR_DEFAULT

Comment (atamazov) [reply CMakeLists.txt:293]: MIOPEN_USE_COMGR is an Cmake option, I need separate variable to initialize its default.

Comment (shurale-nkn) [reply CMakeLists.txt:293]: ohh, yes, right.
Then first case with MIOPEN_USE_COMGR_DEFAULT=true
``If I do so, then I will need to first set MIOPEN_USE_COMGR_DEFAULT to TRUE and then change it to FALSE in some cases. This makes it harder to track the value of MIOPEN_USE_COMGR_DEFAULT. Do you agree?``

Comment (shurale-nkn) [reply CMakeLists.txt:293]: I want this, just not to lose the reason for these exceptions in the code

Comment (atamazov) [reply CMakeLists.txt:293]: But the reason is the same - it doesn't work with OpenCL BE and we do not need to for NOGPU ;)

Comment (atamazov) [reply CMakeLists.txt:293]: I mean that the reason is programmatically expressed in the code. Either in some local place or scattered around. I would prefer locality.

Let's look at it differently. If this is written in C++, then single initialization of MIOPEN_USE_COMGR_DEFAULT allows defining it  as `const`. But with IFs this is not possible.

Comment (shurale-nkn) [reply CMakeLists.txt:293]: For me local place - in backend handlers, afther parametrs check and error message.
But these is only my preferences. I have already approved this PR.

**Comment by atamazov** [comment]
@JehandadKhan I need your blessing))

**Review by JehandadKhan — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**
